### PR TITLE
gh-85046: Adding missing errno system symbols descriptions

### PR DESCRIPTION
**Pull request by YoSTEALTH**
https://bugs.python.org/issue40869

* Issue: gh-85046

### CODE REVIEW

**Comment by YoSTEALTH** [comment]
1. This patch could be back-ported.
2. The source for `errno.h` is at https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/asm-generic/errno.h?h=v5.7

File: Doc/library/errno.rst [hunk 1]
```diff
@@ -637,3 +637,58 @@ defined by the module.  The specific list of defined symbols is available as
...
+
+   State not recoverable
+
+
+.. data:: ERFKILL
+
+   Operation not possible due to RF-kill
+
+
+.. data:: EHWPOISON
```
Comment (remilapeyre) [inline Doc/library/errno.rst:722]: This one is not present in `Modules/errnomodule.c`

**Review by remilapeyre — CHANGES_REQUESTED** [review]
Thanks for adding some, the following ones are still missing as far as I can tell:

- EPROCLIM
- WSAENOTEMPTY
- WSAEHOSTDOWN
- WSAEADDRINUSE
- WSAECONNABORTED
- WSABASEERR
- WSAEPFNOSUPPORT
- WSAHOS
- WSAVERNOTSUPPORTED
- WSAECONNREFUSED
- WSAETIMEDOUT
- WSAEADDRNOTAVAIL
- WSAEBADF
- WSAEALREADY
- WSAEISCONN
- WSAEUSERS
- WSANOTINITIALISED
- WSAEREMOTE
- WSAEINTR
- WSAEPROTOTYPE
- EBADARCH
- WSAEPROCLIM
- EPROGUNAVAIL
- WSAENOPROTOOPT
- WSAGETSELECTERRO
- WSAGETASYNCERRO
- EBADEXEC
- WSAEFAULT
- WSAELOOP
- WSAGETASYNCBUFLE
- WSAEMFILE
- WSAEDISCON
- ENOTACTIVE
- EAUTH
- WSAESOCKTNOSUPPORT
- EPROCUNAVAIL
- WSAMAKEASYNCREPL
- WSAEDQUOT
- ENEEDAUTH
- WSAENOBUFS
- WSAETOOMANYREFS
- WSAEOPNOTSUPP
- WSAENOTSOCK
- WSASYSNOTREADY
- WSAEPROTONOSUPPORT
- ENOTSUP
- WSAMAKESELECTREPL
- ELOCKUNMAPPED
- EBADMACHO
- WSAN
- WSAENAMETOOLONG
- WSAEMSGSIZE
- ENOPOLICY
- WSAEINPROGRESS
- WSAENETUNREACH
- WSAEACCES
- WSAENOTCONN
- EBADRPC
- EFTYPE
- WSADESCRIPTIO
- ERPCMISMATCH
- ENOATTR
- WSAECONNRESET
- WSAESTALE
- WSAGETSELECTEVEN
- EPWROFF
- WSASY
- WSATR
- WSAEINVAL
- WSAESHUTDOWN
- WSAENETRESET
- EDEVERR
- EPROGMISMATCH
- WSAENETDOWN
- WSAEDESTADDRREQ
- ESHLIBVERS
- WSAEAFNOSUPPORT
- WSAEWOULDBLOCK
- WSAEHOSTUNREACH

File: Doc/library/errno.rst (hunk 1 above)
Comment (YoSTEALTH) [reply Doc/library/errno.rst:722]: thanks, added `EHWPOISON`

**Comment by YoSTEALTH** [comment]
@remilapeyre  I think the reason those missing errors were not added is because they are alias for linux errno, for e.g. windows `WSAEWOULDBLOCK` is same as linux `EWOULDBLOCK` and their description are the same.

**Comment by remilapeyre** [comment]
> @remilapeyre I think the reason those missing errors were not added is because they are alias for linux errno, for e.g. windows `WSAEWOULDBLOCK` is same as linux `EWOULDBLOCK` and their description are the same.

What about `EAUTH` for example? Shouldn't it be in the list?

**Comment by YoSTEALTH** [comment]
@remilapeyre Yes, i was just looking at those and other os specific `errno`, i would have to add those.

**Comment by YoSTEALTH** [comment]
I also tried adding missing windows error description but they are not really that useful to users like e.g. `WSAVERNOTSUPPORTED` the description is `Error WSAVERNOTSUPPORTED`

**Comment by remilapeyre** [comment]
> WSAVERNOTSUPPORTED

I don't know Windows well but I think you will find more information in the documentation e.g. https://docs.microsoft.com/en-us/windows/win32/winsock/windows-sockets-error-codes-2

Could we add subsections like "Those errors are specific to MacOS:"? It would break the long list and make it easier to ignore the parts of the list not applicable to our current platform.

**Comment by YoSTEALTH** [comment]
@remilapeyre I have updated what description i could find. I have been adding ``.. availability:: Windows.`` so people know its only for that os. These errors i can't find any description for:

WSATR
WSABASEERR
WSAHOS
WSAGETSELECTERRO
WSAGETASYNCERRO
WSASY
WSAMAKESELECTREPL
WSADESCRIPTIO
WSAN
WSAGETSELECTEVEN
WSAGETASYNCBUFLE
WSAMAKEASYNCREPL

**Comment by YoSTEALTH** [comment]
Core developer reviewer note:
There are duplicate defines like
https://github.com/python/cpython/blob/f69b2b983b792f012846e4bcdb0e578bc8746162/Modules/errnomodule.c#L826
https://github.com/python/cpython/blob/f69b2b983b792f012846e4bcdb0e578bc8746162/Modules/errnomodule.c#L856
should i remove one?

**Review by remilapeyre — APPROVED** [review]
Thanks for making the changes @YoSTEALTH ! A core developer will make another review and answer your question.

**Comment by taleinat** [comment]
> Core developer reviewer note:
> There are duplicate defines like
> 
> https://github.com/python/cpython/blob/f69b2b983b792f012846e4bcdb0e578bc8746162/Modules/errnomodule.c#L826
> 
> https://github.com/python/cpython/blob/f69b2b983b792f012846e4bcdb0e578bc8746162/Modules/errnomodule.c#L856
> 
> should i remove one?

If the first one in is the "main" list, then IMO yes, remove the second one.

**Comment by taleinat** [comment]
> I have been adding `.. availability:: Windows.`

That's rather verbose, but okay IMO on a page like this which is a very technical list.

I'd still suggest breaking up the already very long list, similar to how it is in the source code file, putting OS-specific codes in separate sub-sections.

**Comment by taleinat** [comment]
With the addition of `EHWPOISON`, this is no longer documentation-only.

We don't detail doc-only changes in NEWS, so the NEWS entry should only mention the addition of `EHWPOISON`.

**Comment by YoSTEALTH** [comment]
@taleinat ok, thanks for your comments, I will get to these changes on/before this coming weekend.

**Comment by YoSTEALTH** [comment]
Could this be merged as is for now? Will create new PR for other changes.

**Comment by furkanonder** [comment]
@YoSTEALTH Could you solve the conflicts?

**Comment by YoSTEALTH** [comment]
I am not sure whats going on. I just don't have the time + proper setup to deal with coding issues anymore. If this issues doesn't get resolved soon, I am just going to close this PR, as is been many years.

**Pull request closed without merging**